### DISCUSSION TITLE
opencv on android

### DISCUSSION THREAD

**Discussion by vbnmjj** [beeware/toga#2505] [Ideas]
I see a hardware example  ,This means that if I execute the following code, Can I use the camera of my mobile phone by opencv?

class ExampleHardwareApp(toga.App):
    def startup(self):
        try:
            # This will provide a prompt for camera permissions at startup.
            # If permission is denied, the app will continue.
            self.camera.request_permission()
        except NotImplementedError:
            print("The Camera API is not implemented on this platform")

**Comment by freakboy3742** [comment]
It's a little unclear exactly what you're asking.

Toga's camera API allows you to capture an image using iOS, Android, or macOS camera. If you want to pass that image to OpenCV, then you can do so, although you'll need to convert the `toga.Image` object into a format compatible with OpenCV.

However, Toga gaining a camera API doesn't add any features to OpenCV. I believe OpenCV may have some native camera APIs; however, I know nothing about them, nor do I have any reason to believe that Toga adding a camera API would have any impact on OpenCV's camera support (as OpenCV doesn't have any dependency on Toga). If you want to know if you can use OpenCV's camera API, you'd need to direct that question to OpenCV.
